reactor-core/src/main/java/reactor/core/publisher/Mono.java

```diff
@@ -2216,7 +2222,9 @@ public T block(Duration timeout) {
 	 */
 	public final Mono<T> doOnError(Consumer<? super Throwable> onError) {
 		Objects.requireNonNull(onError, "onError");
-		return doOnSignal(this, null, null, onError, null, null, null);
+		return doOnSuccessOrError((ignoreValue, error) -> {
```

Comment:
Do we need to change doOnError , if we observe an error in a mono flow it's already too late for propagating any other signal ?

Reply:
actually, this one should be changed to `doOnTerminalSignal(..., onError)`: no allocation for the individual handlers (success or error) in the new version of `MonoPeekTerminal`. I included this change because it aligns all onComplete/onError signals into a single operator.